neonvm/tools/vm-builder-generic/main.go

```diff
@@ -211,23 +211,33 @@ func printReader(reader io.ReadCloser) error {
 	return nil
 }
 
-func AddToTar(tw *tar.Writer, filename string, buf bytes.Buffer) error {
+func AddTemplatedFileToTar(tw *tar.Writer, buf *bytes.Buffer, tmplData any, filename string, tmplString string) error {
+	tmpl, err := template.New(filename).Parse(tmplString)
+	if err != nil {
+		return fmt.Errorf("failed to parse template for %q: %w", filename, err)
+	}
+
+	buf.Reset()
```

Comment:
why pass in a buffer that we're just going to reset and then write from?

Reply:
The idea behind keeping the buffer around is that the backing slice's capacity is reused. It's.... mildly more efficient, and in-line with the previous implementation.

That said, I think (a) the performance benefit is minimal and (b) the weirdness is significant. Removed it.